server-utils/src/stream_codec.rs

```diff
@@ -87,12 +87,14 @@ impl Decoder for StreamCodec {
 				else if byte == b'"' && !is_escaped {
 					in_str = !in_str;
 				}
-				else if byte == b'\\' && is_escaped && !in_str {
-					is_escaped = !is_escaped;
-				}
 				else if is_whitespace(byte) {
 					whitespaces += 1;
 				}
+				if byte == b'\\' && !is_escaped && in_str {
```

Comment:
should be started with `else` as well

Reply:
I didn't use `else if` here, as I wanted `is_escaped` to be unset in all cases. With `else if`, it wouldn't be unset if a whitespace followed the backslash. That said, it wouldn't be correctly formed JSON if any of the other cases are caught.

Reply:
Oh, you're absolutely right. Could you add more test cases covering these cases?